### PR TITLE
[IMP] web: redesign toaster

### PR DESCRIPTION
**Pull request by bvr-odoo**
Remove the icon of the toaster. Remove the header and move the title in
the body. Move buttons at the bottom instead of at the right side.

task-2311554

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/widgets/notification.js
Comment (qsm-odoo) [inline addons/web/static/src/js/widgets/notification.js:71]: The task is about making the toast looks like in the backend but you are modifying the framework code here... does it mean no backend toast was using those options maybe?

File: addons/web/static/src/xml/notification.xml
Comment (qsm-odoo) [inline addons/web/static/src/xml/notification.xml:21]: That class does not exist

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/web/static/src/js/widgets/notification.js
Comment (bvr-odoo) [reply addons/web/static/src/js/widgets/notification.js:71]: No, some backend toast use also those options ...
So this PR remove the icons also in backend toast (when it's used)... But logically it sould be removed everywhere, if we dont like it 
@sbuhl  right ?

Comment (bvr-odoo) [reply addons/web/static/src/js/widgets/notification.js:71]: @sbuhl What do you thing ?

Comment (sbuhl) [reply addons/web/static/src/js/widgets/notification.js:71]: @bvr-odoo 
We want to get rid of them in the backend as well (I though it was already the case), so I'm ok with your removal.

I notify @aab-odoo on a precautionary basis (already got the green light from NGR).

cc @qsm-odoo

Comment (aab-odoo) [reply addons/web/static/src/js/widgets/notification.js:71]: @sbuhl Thanks for the ping, we'll adapt the code accordingly in our "wowl refactoring" branch (where we rewrite from scratch the webclient and among other things, the notification system).

**Review by qsm-odoo — APPROVED** [review]
@robodoo r+

**Review by qsm-odoo — APPROVED** [review]
@robodoo r+

**Comment by qsm-odoo** [comment]
Hello @odoo/rd-security 
We are removing a t-raw and moving two existing ones around here, can you make this move forward? Thank you!

**Comment by xmo-odoo** [comment]
This seems like a good occasion to fix the entire thing and remove the t-raws entirely no?

**Comment by qsm-odoo** [comment]
@xmo-odoo 
If I am not mistaken @aab-odoo told they will be refactoring that code with OWL in the next months, maybe we could leave that to them? But by curiosity, how would you remove the t-raw here?

**Comment by aab-odoo** [comment]
@qsm-odoo Indeed. It's already done, actually, and the t-raw's are still there...

**Comment by xmo-odoo** [comment]
> @xmo-odoo
> If I am not mistaken @aab-odoo told they will be refactoring that code with OWL in the next months, maybe we could leave that to them? But by curiosity, how would you remove the t-raw here?

Maybe by changing the protocol and saying that the notification takes markdown (or something) instead of raw HTML, then adding a markdown widget for t-esc? Are t-esc widgets a thing client-side?

**Comment by xmo-odoo** [comment]
(also it's "toast", because the notification scolling into view looks a bit like a toast popping out, a toaster is something you put bread in).

**Comment by JKE-be** [comment]
markdown in toasts ... 
Curious to see all craps needed to adapt this toast mechanism with case where you have button to do action, and co ...
Do we plan to extend markdown to support a lot of new option ? Or just say 'no way' when PO ask any update and restrict our dev to what markdown allow. In this case maybe we could just work in PS tech and don't call our department RD...

While I'm sure that markdown in forum will be a cool feature... not sure about toast...

So, In case remove a icon and move a t-raw withtout change behaviour is blocked. We can just close it.

**Comment by xmo-odoo** [comment]
> Curious to see all craps needed to adapt this toast mechanism with case where you have button to do action, and co ...

Are there actually notifications which use buttons at the moment? That seems way the hell out there in terms of notification features. IIRC in standard there's only some basic formatting and a few links, hence the suggestion of markdown possibly being suitable for that use case as an alternative to the wide-open XSS vector that notifications currently are.

> Do we plan to extend markdown to support a lot of new option ? Or just say 'no way' when PO ask any update and restrict our dev to what markdown allow. In this case maybe we could just work in PS tech and don't call our department RD...

:question:

**Comment by JKE-be** [comment]
image

And yes we could imagine to drop the support of html in toast (maybe time to do it in owl rewriting to avoid to reinvent a wheel now for nothing ? ) like for modal title... 
But it is really not the goal of this PR that is just to remove the icon and header zone, to make Usability better.
It is not like if we were introduce new feature, or anything else... Owl is already rewriting all features including this one.

So Yes, disappointed to ask to rewrite all the notifications from all odoo code, to remove the title on each toast, and add the title inside the content manually. and add new CSS code just to hide an icon loaded for nothing... Just because it is the way to have exactly the same usability, without move the t-raw and so have the PR blocked for no good reason.

At the end, we will have a big diff, all translation will be lost, we will have extra css, ... Imo the change done in this PR was the best solution to avoid it.

**Pull request closed without merging**
